enhancements/update/cvo-techpreview-manifests.md

```diff
@@ -0,0 +1,139 @@
+Since most people add manifests in order to see them applied, they will notice if their manifest is never created and
+the choice of annotation allows for potential future usage in the CVO consistent with feature gate handling used by
+other operators.
+
+If a manifest sets `.metadata.annotations["config.openshift.io/feature-gate"]="TechPreviewNoUpgrade"`, the manifest will
+not be created unless `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"`.
+If `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"`, then the manifest is reconciled as normal
+for whatever stage the CVO is in.
+The `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"` can be included in the installation
+manifests provided by a customer (we do this some CI scenarios as well), but there is no expected CVO change for bootstrapping.
+If bootstrapping changes are required, it will be responsibility of the contributing team to manage the proper inclusion
+in the installer and other bootstrapping components.
+
+Because the cluster cannot upgrade to 4.y+1 and because the TechPreviewNoUpgrade flag cannot be unset, there is no concern
+about removing any manifests that were added for TechPreviewNoUpgrade. 
+
+### User Stories
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+1. The TechPreview CI jobs (already present), should run after this feature is implemented.
+2. In 4.10, the cluster-api clusteroperator should not be present in our "normal" CI flows. 
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+When set, `TechPreviewNoUpgrade` prevents al 4.y to 4.y+1 upgrades.
+
+### Version Skew Strategy
+
+Because upgrades are prevented when this feature is active, there is no supported skew.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+### Inert Operators
+It is possible for every tech preview operator to
+1. create an inert run mode
+2. self-bootstrap resources "normally" created by the CVO, like CRDS.
+3. if the operator doesn't graduate, it must continue to be included in the payload and list its CVO managed resources
+    to remove them in the next release.
+5. create a clusteroperator that every customer has ignore
+
+This fans the problem out from a single point (CVO), to every tech preview operator.
+It also means that a cluster-admins have to ignore a clusteroperator in the list that is tech preview. 
```

Comment:
I'm not clear on why this ignoring is a burden.  Is it something we need to fix for the image registry operator's [`managementState: Unmanaged`][1]?

[1]: https://docs.openshift.com/container-platform/4.9/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator

Reply:
> I'm not clear on why this ignoring is a burden. Is it something we need to fix for the image registry operator's [`managementState: Unmanaged`](https://docs.openshift.com/container-platform/4.9/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator)?

The image-registry operator is not a techpreview operator.  It is fully supported. When you search for it, you find docs, if it has a problem you know you are fully supported.  Seeing a techpreview operator on your not techpreview cluster doesn't seem comparable to me.

Reply:
ah, so the concern isn't:

> huh, a ClusterOperator for `$COMPONENT`?  I didn't think I had that installed...

We're ok with that conflation between the inert-mode operator and its backing operands.  The concern is:

> huh, a ClusterOperator for `$COMPONENT`?  I wonder what that does... Getting the docs... What?! `TechPreviewNoUpgrade`!  My cluster is doomed!

Can we be more specific in the wording?  Something like:

> It also increases the chance that a cluster admin would misinterpret an inert-mode operator as tainting their cluster with unsupported, tech-preview bits.

Reply:
One thing to note, Tech Preview features should always be documented.